Lib/sre_constants.py

```diff
@@ -21,6 +21,17 @@
 # should this really be here?
 
 class error(Exception):
+    '''Exception raised for invalid regular expressions.
```

Comment:
It is also raised by `Pattern.sub()` and `Match.expand()` when the replacement string contains invalid group reference. E.g. `re.compile(r'\w').sub(r'\1', 'a')`, `re.match(r'\w', 'a').expand(r'\1')`. Technically it is raised not during compilation, but immediately after it from user's point of view.